scipy/optimize/tests/test_minpack.py

```diff
@@ -551,6 +551,73 @@ def test_NaN_handling(self):
         assert_raises(ValueError, curve_fit, lambda x, a, b: a*x + b,
                       xdata, ydata, **{"check_finite": True})
 
+    @staticmethod
+    def _check_nan_policy(f, xdata_with_nan, xdata_without_nan,
+                          ydata_with_nan, ydata_without_nan, method):
+        kwargs = {'f': f, 'xdata': xdata_with_nan, 'ydata': ydata_with_nan,
+                  'method': method, 'check_finite': False}
+        # propagate test
+        error_msg = ("`nan_policy='propagate'` is not supported "
+                     "by this function.")
+        with assert_raises(ValueError, match=error_msg):
+            curve_fit(**kwargs, nan_policy="propagate", maxfev=2000)
+
+        # raise test
+        with assert_raises(ValueError, match="The input contains nan"):
+            curve_fit(**kwargs, nan_policy="raise")
+
+        # omit test
+        result_with_nan, _ = curve_fit(**kwargs, nan_policy="omit")
+        result_without_nan, _ = curve_fit(**kwargs, nan_policy="omit")
```

Comment:
I think you forgot to use `xdata_without_nan` and ``ydata_without_nan`` here. Shouldn't it be:

```suggestion
        result_with_nan, _ = curve_fit(**kwargs, nan_policy="omit")
        kwargs['xdata'] = xdata_without_nan
        kwargs['ydata'] = ydata_without_nan
        result_without_nan, _ = curve_fit(**kwargs, nan_policy="omit")
```

? Otherwise, the test doesn't really test anything...

Reply:
Also, we can avoid passing `nan_policy='omit'` argument in the second call to `curve_fit`